### DISCUSSION TITLE
Receiving (not pulling or writing) data from  AB Compact GuardLogix PLC

### DISCUSSION THREAD

**Discussion by dev-svk-flbs** [ottowayi/pycomm3#157] [Q&A]
Hello, 

We are trying to RECEIVE data from a AB Compact GuardLogix PLC  over ethernet IP. 
Most examples of pycomm3 usage shows that the communication always initiates from the host computer running pycomm3 (a raspberry pi, for example), pycomm3 either pulls or writes data from or to a PLC. 
This does not apply for our application. 
We need to receive data from a already programmed Compact GuardLogix PLC to read 5/6 different tags sent ~every minute.  
It seems like we need to have some sort of listener running on the pi to listen to incoming data from the PLC over ethernet IP. 
Is this something we can do using pycomm3? If not, can anyone please point me to the right direction?

Thanks.

**Comment by ottowayi** [comment]
Currently that is not a part of pycomm3, but there is an issue open to investigate adding it.  I would like to say it will be added, but I can't give any timeframe since I haven't even started looking into it.   You should look into [cpppo](https://github.com/pjkundert/cpppo), I believe it supports what you need.
